pkg/kubelet/kubelet_getters.go

```diff
@@ -240,12 +240,25 @@ func (kl *Kubelet) GetNode() (*v1.Node, error) {
 
 // getNodeAnyWay() must return a *v1.Node which is required by RunGeneralPredicates().
 // The *v1.Node is obtained as follows:
-// Return kubelet's nodeInfo for this node, except on error or if in standalone mode,
+// Return kubelet's nodeInfo for this node. It also checks if the OS and Arch labels are mismatched, if yes
+// they're updated and the new node object with modified values is returned, if not the object is returned as is.
+// except on error or if in standalone mode,
 // in which case return a manufactured nodeInfo representing a node with no pods,
 // zero capacity, and the default labels.
 func (kl *Kubelet) getNodeAnyWay() (*v1.Node, error) {
 	if kl.kubeClient != nil {
 		if n, err := kl.nodeLister.Get(string(kl.nodeName)); err == nil {
+			// We assume that the node labels match the host OS and arch but in order to avoid situations
+			// where the OS label was modified and kubelet has not synced the labels yet update it once here.
+			// This method is called by PredicateAdmitHandler.
+			if runtime.GOOS != n.Labels[v1.LabelOSStable] || runtime.GOARCH != n.Labels[v1.LabelArchStable] {
+				node, err := kl.syncOSAndArchLabelsOnce()
+				if err != nil {
+					return nil, err
+				}
+				// Get the updated node object.
+				n = node.DeepCopy()
```

Comment:
Why a deep copy and not a reassignment? We also already appear to be making a deep copy inside of `syncOSAndArchLabelsOnce`.

Reply:
I prefer deepcopies to shallowcopies for structs. The main reason is structs can evolve over time and shallow copies unless tested thoroughly can be error prone. 

The reason for having deepcopy in `syncOSAndArchLabelsOnce` is same. When we call `syncOSAndArchLabelsOnce` from `syncOSAndArchLabels`, in order to do patching we are making a copy.